src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -287,6 +290,27 @@ public IActionResult GetAdminToken()
             return Ok(requestHeaderToken);
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            FunctionDescriptor function = scriptHost.GetWarmupFunctionOrNull();
```

Comment:
Nice. Now you could move the rest of this method into your new helper extension class, so you could simply do `scriptHost.InvokeWarmupAsync()` or perhaps `bool scriptHost.TryInvokeWarmupAsync()`. That keeps the controller method clean, and makes the logic easily testable.